### PR TITLE
[XLA:GPU] terminology change in GpuExecutable

### PR DESCRIPTION
**Pull request by whchung**
To cope with upcoming introduction of AMDGPU support in XLA, change terms used
in GpuExecutable.

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/nvptx_executable.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#include "tensorflow/compiler/xla/types.h"
+#include "tensorflow/core/lib/core/stringpiece.h"
+#include "tensorflow/core/lib/gtl/array_slice.h"
+#include "tensorflow/core/platform/macros.h"
+#include "tensorflow/core/platform/stream_executor_no_cuda.h"
+
+namespace xla {
+namespace gpu {
+
+class NVPTXExecutable : public GpuExecutable {
```
Comment (jlebar) [inline tensorflow/compiler/xla/service/gpu/nvptx_executable.h]: All classes must have a (brief OK) description in the header file.

File: tensorflow/compiler/xla/service/gpu/gpu_executable.cc [hunk 2]
```diff
@@ -46,18 +46,16 @@ using tensorflow::tracing::ScopedAnnotation;
 // Implementation note: HLO profiling is always enabled for GPU executables,
 // since we can use timers around thunks.
 GpuExecutable::GpuExecutable(
-    const string& ptx, const std::vector<uint8>& cubin,
-    std::pair<int, int> compute_capability,
+    const string& text, const std::vector<uint8>& binary,
```
Comment (jlebar) [inline tensorflow/compiler/xla/service/gpu/gpu_executable.cc]: "text" is kind of generic.  Could we call it "assembly_text" or "asm_text"?

**Review by jlebar — CHANGES_REQUESTED** [review]
It seems like all of the complexity here stems from the fact that the way we represent a hardware version is different between NVGPU and AMDGPU.  In NVGPU it's a pair<int,int>, and in AMDGPU, it's a plain int.

It seems to me that an alternative that avoids a lot of complexity would be to represent the AMDGPU version as a pair<int, int>, where the second value is 0.  Or if you like, it could be a struct/class.

File: tensorflow/compiler/xla/service/gpu/nvptx_executable.h (hunk 1 above)
Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/nvptx_executable.h]: will do

File: tensorflow/compiler/xla/service/gpu/gpu_executable.cc (hunk 2 above)
Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/gpu_executable.cc]: let me revise it as `asm_text`

**Comment by whchung** [comment]
@jlebar I think it's fine to use `std::pair<int, int>` for AMDGPU LLVM backend and leave the second value be 0.

However there is another complexity imposed in the interface of `DeviceDescription::cuda_compute_capability()`. Certainly it's not ideal to call such interface in a generic `GpuExecutable`.

In PR #20709 , I amended an member function `DeviceDescription::rocm_amdgpu_isa_version()` for ROCm path, and let AMDGPU subclass of `GpuExecutable`, `AMDGPUExecutable` , use this new interface.

Do you see it feasible to merge these 2 interfaces into one in StreamExecutor and give is a more generic name?

**Comment by jlebar** [comment]
> Do you see it feasible to merge these 2 interfaces into one in StreamExecutor and give it a more generic name?

Yes.  And/or you could provide a third function.

**Comment by whchung** [comment]
@jlebar It turns out the rabbit hole is quite deep. In order to ditch `std::pair<int, int>` in XLA, and adopt a common `DeviceVersion` class for NVPTX and AMDGPU, the following components have to be changed:

- GPU common runtime : move `CudaVersion` to StreamExecutor and rename it to `DeviceVersion`
- StreamExecutor : instead of using `cc_major` and `cc_minor`, use `DeviceVersion`
- XLA : forefeit interfaces which take `std::pair<int, int>` and switch to `DeviceVersion`

I'll submit one PR for this soon and will then revise this PR.

**Review by jlebar — DISMISSED** [review]
Approve the top commit of this PR, will look at the other commit in the separate PR.

**Comment by whchung** [comment]
rebase after changing #20786

**Comment by tensorflowbutler** [comment]
Nagging Assignee @qlzh727: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by qlzh727** [comment]
Not sure if this PR is still alive or not. Please resolve the merge conflict before proceeding.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**